src/system/SystemLayerImplFreeRTOS.h

```diff
@@ -29,7 +29,11 @@
 namespace chip {
 namespace System {
 
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS_PLATFORM
+class LayerImplFreeRTOS : public LayerSockets
+#else
 class LayerImplFreeRTOS : public LayerFreeRTOS
+#endif // CHIP_SYSTEM_CONFIG_USE_SOCKETS_PLATFORM
```

Comment:
This isn't the right way to implement this - We should create a seperate class `LayerImplSocketFreeRTOS` that inherits from both to leverage the current implementation and the functions of the `LayerSockets`.

Reply:
created a new class